native/src/main/java/io/ballerina/stdlib/xmldata/AttributeManager.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.xmldata;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Attribute manager handles attributes and namespaces of XML elements.
+ *
+ * @since 2.0.1
+ */
+public class AttributeManager {
+
+    private ConcurrentHashMap<String, String> concurrentHashMap;
```

Comment:
Use LinkedHashMap here, since AttributeManager is concurrent safe now. Also we need to preserve the order, so replace other usages with LinkedHashMap as well.

Reply:
Fixed in 3481ab2